src/Compiler/pars.fsy

```diff
@@ -2010,7 +2010,7 @@ inheritsDefn:
   | INHERIT ends_coming_soon_or_recover
      { let mDecl = (rhs parseState 1)
        if not $2 then errorR(Error(FSComp.SR.parsTypeNameCannotBeEmpty(), mDecl))
-       SynMemberDefn.Inherit(SynType.LongIdent(SynLongIdent([], [], [])), None, mDecl) }
+       SynMemberDefn.Inherit(SynType.LongIdent(SynLongIdent([], [], []), false), None, mDecl) }
```

Comment:
The fact that you need to pass `false` in so many locations is a bit of a code smell to me.
Would at this point a new grammar rule not make more sense?

Reply:
I'm not really following, if you were to write `f<module2 M, int> ()` today ([link](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWTwA8AtuwFcasAExgAsrjAoALgD4wACgCUkEAF8gA)), you get exactly a `SynType.App` for `module2 M`.

And regardless of how your model this in `SynType`, you will find theoretical scenario's where you can plug in `module X` wherever a `SynType` is allowed.

Reply:
Me neither:). What does your suggestion look like then?

Reply:
But `module A.B.C` is in and of itself not an application. `module` is no type (to be stored in `typeName`), but a disambiguating keyword. Technically it isn't even needed and `LongIdent` could have been reused.

I think we'd just be abusing `SynType.App` and making it confusing.

Reply:
`module A.B.C` is syntactically an application. This is valid jargon for the untyped tree. Just like `int list` is an application in the untyped tree. Of course, there is no `int` function that is being applied by a `list`, but it is a `SynType.App` nonetheless.

The evaluation of `module A.B.C` will later, in the typed tree, of course not be an application, but on the untyped level, I'm calling it as I see it.

Reply:
Ok, I get it now. However, if the parser can already tell `module A.B.C` and `int list` are semantically different, shouldn't they be encoded differently? Wouldn't it be cleaner than inspecting `typeName` for `module` in CheckExpressions.fs?

Reply:
Well, that is part of the debate, is this really different enough to have its own DU case?
From an untyped perspective, I'd say no. Most `SynType` are rather undecided yet on what they actually will represent later in the typed tree.

I think https://github.com/dotnet/fsharp/pull/13700 is a bit of an example. We allowed users to write `int array2d` instead of `int[,]`. This could have been a new case in the `SynType` DU, except it just fits in what is already out there. That feature wasn't new from a syntax point of view.

And just because you are using the `module` keyword, I'm still convinced this is similar.

You have my two cents, ultimately it is not up to me. The maintainers will tell you what they want eventually. If they approve your current code, I would implore you to include the full range of the type, as that will be problematic sooner or later.

Lastly, I do want to mention that this is a solid PR and you have done amazing work here.
I may seem a tad too critical in reviewing this, but that is only because I have a stake in the `SyntaxTree`. Every change made to it affects me and I will voice my concerns when I deem it necessary. I expect no different from my peers when they review my PRs.
Keep it up!

Reply:
>  I would implore you to include the full range of the type

Yeah, I have already done this locally.

> I may seem a tad too critical in reviewing this

No, no, I love thorough PR scrutiny and discussing the implementation in order to maintain high standards, especially in a project like the F# compiler. So thanks for taking the time to review this.